oomact_ros/src/feeder/ImuInputFeeder.cpp

```diff
@@ -0,0 +1,46 @@
+#include <aslam/calibration/data/ObservationManagerI.h>
+#include <aslam/calibration/data/MeasurementsContainer.h>
+#include <aslam/calibration/data/AccelerometerMeasurement.h>
+#include <aslam/calibration/data/GyroscopeMeasurement.h>
+#include <aslam/calibration/input/InputReceiverI.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <sensor_msgs/Imu.h>
+
+#include "aslam/calibration/ros/InputFeederFactoryRegistry.h"
+#include "aslam/calibration/ros/InputFeederFactoryI.h"
+#include "aslam/calibration/ros/InputFeederTemplates.h"
+
+namespace aslam {
+namespace calibration {
+namespace ros {
+
+bool msg2Measurement(const sensor_msgs::Imu &msg, AccelerometerMeasurement & m){
+  if(msg.linear_acceleration_covariance[0] == 0.0){
+    return false;
+  }
+  m.a = rosVector3dToEigenVector3(msg.linear_acceleration);
```

Comment:
Having run into Eigen alignment issues recently I have been on the lookout for them, and I feel you may hit issues with some of your measurements. 
For example this one https://github.com/ethz-asl/oomact/blob/master/oomact/include/aslam/calibration/data/PoseMeasurement.h#L13 is unsafe due to the Eigen::Vector4d. It also seems from Ubuntu 16.04 (due to having Eigen 3.3) that Eigen::Vector3d is also vectorized which can make things much more difficult.

Reply:
Yes, this is already on my todo list. One Idea is to go away from Eigen vectors as data storage or use unaligned versions. I don't like the overhead in memory for something that almost all the time gets only copied eventually.